src/HealthGPS/kevin_hall_model.cpp

```diff
@@ -195,53 +253,110 @@ void KevinHallModel::set_energy_intake(Person &person) const {
     }
 }
 
-SimulatePersonState KevinHallModel::simulate_person(Person &person, double shift) const {
-    // Initial simulated person state.
-    const double H_0 = person.get_risk_factor_value("Height"_id);
-    const double BW_0 = person.get_risk_factor_value("Weight"_id);
-    const double PAL_0 = person.get_risk_factor_value("PhysicalActivity"_id);
-    const double F_0 = person.get_risk_factor_value("BodyFat"_id);
-    const double L_0 = person.get_risk_factor_value("LeanTissue"_id);
-    const double ECF_0 = person.get_risk_factor_value("ExtracellularFluid"_id);
-    const double G_0 = person.get_risk_factor_value("Glycogen"_id);
-    const double EI_0 = person.get_risk_factor_value("EnergyIntake"_id);
-    const double CI_0 = person.get_risk_factor_value("Carbohydrate"_id);
+void KevinHallModel::initialise_kevin_hall_state(Person &person,
+                                                 std::optional<double> adjustment) const {
 
-    // TODO: Compute height.
-    double H = H_0;
+    // Apply optional weight adjustment.
+    if (adjustment.has_value()) {
+        person.risk_factors.at("Weight"_id) += adjustment.value();
+    }
+
+    // Get already computed values.
+    double H = 0; // TODO: person.risk_factors.at("Height"_id);
+    double BW = person.risk_factors.at("Weight"_id);
+    double PAL = person.risk_factors.at("PhysicalActivity"_id);
+    double EI = person.risk_factors.at("EnergyIntake"_id);
+
+    // Body fat.
+    double F;
+    if (person.gender == core::Gender::female) {
+        F = BW * (0.14 * person.age + 39.96 * log(BW / pow(H / 100, 2.0)) - 102.01) / 100.0;
+    } else if (person.gender == core::Gender::male) {
+        F = BW * (0.14 * person.age + 37.31 * log(BW / pow(H / 100, 2.0)) - 103.94) / 100.0;
+    } else {
+        throw core::HgpsException("Unknown gender");
```

Comment:
Is it possible to get to this point of the simulation with an unknown gender?

Reply:
In theory, no, but I left it in to be safe. What do you do with code paths you expect not to be reached yourself?

Reply:
No, I think it is good practice to handle all possible cases, specially in Python. However, in here, the compiler will not let you assign `person.gender` anything other than `core::Gender::male` or `core::Gender::female`, so I'm not sure it this is even a possible case.

Reply:
If someone was being really perverse I *think* they could do something like `static_cast<core::Gender>(100)` to get a non-male, non-female value, but I don't think we should worry about that case. It's more of a concern when you are casting a user-supplied number to an `enum`, but that doesn't apply here.

Reply:
There are other functions dotted around which construct new, or reinitialise or otherwise change `Person`s. As you say, not worth the risk.